internal/pkg/build/oci/oci.go

```diff
@@ -88,38 +98,64 @@ func (t *ImageReference) newImageSource(ctx context.Context, sys *types.SystemCo
 // ParseImageName parses a uri (e.g. docker://ubuntu) into it's transport:reference
 // combination and then returns the proper reference
 func ParseImageName(ctx context.Context, imgCache *cache.Handle, uri string, sys *types.SystemContext) (types.ImageReference, error) {
-	ref, err := parseURI(uri)
+	ref, _, err := parseURI(uri)
 	if err != nil {
 		return nil, fmt.Errorf("unable to parse image name %v: %v", uri, err)
 	}
 
 	return ConvertReference(ctx, imgCache, ref, sys)
 }
 
-func parseURI(uri string) (types.ImageReference, error) {
+func parseURI(uri string) (types.ImageReference, string, error) {
 	sylog.Debugf("Parsing %s into reference", uri)
 
+	arch := ""
 	split := strings.SplitN(uri, ":", 2)
 	if len(split) != 2 {
-		return nil, fmt.Errorf("%s not in transport:reference pair", uri)
+		return nil, arch, fmt.Errorf("%s not in transport:reference pair", uri)
 	}
 
 	transport := transports.Get(split[0])
 	if transport == nil {
-		return nil, fmt.Errorf("%s not a registered transport", split[0])
+		return nil, arch, fmt.Errorf("%s not a registered transport", split[0])
+	}
+
+	uriTmp := split[1]
+	uriTmp = strings.TrimPrefix(uriTmp, "//")
+	archTmp := strings.SplitN(uriTmp, "/", 2)
+
+	if len(archTmp) == 2 {
+		if val, ok := archMap[archTmp[0]]; ok {
+			// replace with golang runtime.GOARCH
```

Comment:
I don't understand this comment.

Reply:
Oh, I think you mean "translate into golang's corresponding runtime.GOARCH value"? Is that right?

Reply:
yes, it's mapping the docker image <-> golang arch